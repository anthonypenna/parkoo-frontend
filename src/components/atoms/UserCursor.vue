<template>
  <div v-once class="user-cursor"></div>
</template>

<script lang="ts">
import Vue from "vue";
import { USER_CURSOR_NAME } from "@/constants/user-cursor";

export default Vue.extend({
  name: USER_CURSOR_NAME,

  props: {
    lat: {
      default: 0,
      type: Number,
    },

    lng: {
      default: 0,
      type: Number,
    },
  },
});
</script>

<style lang="scss" scoped>
.user-cursor {
  width: 32px;
  height: 32px;
  background: #ff9979;
  border: solid 4px #f1f1f1;
  border-radius: 50%;
  animation: pulse 1s linear infinite;
}

@keyframes pulse {
  0% {
    box-shadow: 0 0 0 #ff9979;
  }

  80% {
    box-shadow: 0 0 32px #ff9979;
  }

  100% {
    box-shadow: 0 0 0 #ff9979;
  }
}
</style>
