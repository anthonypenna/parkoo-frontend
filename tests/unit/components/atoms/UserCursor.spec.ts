import { shallowMount } from '@vue/test-utils'
import UserCursor from '@/components/atoms/UserCursor.vue'

describe('<UserCursor />', () => {
  it('should render correctly', () => {
    const wrapper = shallowMount(UserCursor)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
