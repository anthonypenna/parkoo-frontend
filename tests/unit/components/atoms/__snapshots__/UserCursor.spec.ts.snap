// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCursor /> should render correctly 1`] = `<div class="user-cursor"></div>`;
